packages/medusa/src/services/order-edit.ts

```diff
@@ -370,6 +392,198 @@ export default class OrderEditService extends TransactionBaseService {
+
+    let clonedLineItem = await lineItemServiceTx.retrieve(lineItemId, {
+      relations: ["variant"],
+    })
+
+    const diffQuantity = data.quantity - clonedLineItem.fulfilled_quantity
+    if (diffQuantity > 0) {
+      await this.inventoryService_
+        .withTransaction(manager)
+        .confirmInventory(clonedLineItem.variant_id, diffQuantity)
+    }
+
+    clonedLineItem = await lineItemServiceTx.update(clonedLineItem.id, {
+      quantity: data.quantity,
+    })
+
+    await this.refreshAdjustmentAndTaxLines(orderEditId, clonedLineItem.id)
+
+    return clonedLineItem
+  }
+
+  protected async cloneOriginalLineItem(
+    orderEditId: string,
+    lineItemId: string,
+    data: { quantity: number }
+  ): Promise<LineItem> {
+    const manager = this.transactionManager_ ?? this.manager_
+
+    const lineItemServiceTx = this.lineItemService_.withTransaction(manager)
+
+    const lineItem = await lineItemServiceTx.retrieve(lineItemId)
+
+    const diffQuantity = data.quantity - lineItem.fulfilled_quantity
+    if (diffQuantity > 0) {
+      await this.inventoryService_
+        .withTransaction(manager)
+        .confirmInventory(lineItem.variant_id, diffQuantity)
+    }
+
+    const clonedLineItemData = {
+      ...lineItem,
+      id: undefined,
+      order_id: undefined,
+      cart_id: undefined,
+      quantity: data.quantity,
+      claim_order_id: undefined,
+      swap_id: undefined,
+    }
+    const clonedLineItem = await lineItemServiceTx.create(clonedLineItemData)
+
+    await this.refreshAdjustmentAndTaxLines(orderEditId, clonedLineItem.id)
+
+    return clonedLineItem
+  }
+
+  protected async refreshAdjustmentAndTaxLines(
+    orderEditId: string,
+    lineItemId: string
+  ): Promise<void> {
+    const manager = this.transactionManager_ ?? this.manager_
+
+    const orderEdit = await this.retrieve(orderEditId, {
+      relations: [
+        "order",
+        "order.cart",
+        "order.cart.customer",
+        "order.cart.discounts",
+        "order.cart.discounts.rule",
+        "order.cart.gift_cards",
+        "order.cart.region",
+        "order.cart.region.tax_rates",
+        "order.cart.shipping_address",
+        "order.cart.shipping_methods",
+        "order.region",
+      ],
+    })
+    const lineItem = await this.lineItemService_
+      .withTransaction(manager)
+      .retrieve(lineItemId)
+
+    const lineItemAdjustmentServiceTx =
+      this.lineItemAdjustmentService_.withTransaction(manager)
+
+    await lineItemAdjustmentServiceTx.delete({
+      item_id: lineItem.id,
+    })
+    await lineItemAdjustmentServiceTx.createAdjustments(
+      orderEdit.order.cart,
+      lineItem
+    )
+
+    // Calculate context only on the current given line item
+    orderEdit.order.items = [lineItem]
+    const calcContext = await this.totalsService_
+      .withTransaction(manager)
+      .getCalculationContext(orderEdit.order, { exclude_shipping: true })
```

Comment:
todo: for the totals service to correctly compute the allocation map it needs all the items that e.g. a discount has to be distributed over. Take the scenario below:

- Order has fixed discount of 100 USD
- There are 5 items in the cart each has the same item total
- The allocation of the 100 USD across the 5 items will be 20 USD for each item

If we only provide a single item, the TotalsService will compute an allocation of 100 USD towards that single item.

---
The issue above poses a larger complexity to this computation than initially thought of. E.g., continuing from the example above imagine:
- An OrderEdit is made that adds 5 additional items to the order (again with the same item total)
- Do we recompute the line item adjustments so that all items have a discount of 10 USD or do we stick with the original 20 USD on the old items and then make new items charged at the full price?

My suggestion as to what to do here would be to modify the  `createAdjustments` method to be able to filter out `fixed` discounts. This should result in them only being applied when the order is placed.

---
To clarify what this means when **when updating an item**, the expected behavior is this:

Original Order:
- Discount (100 USD fixed)
- Items:
  - Item 1 - Qty: 1, Total 50 USD, Discount 20 USD
  - Item 2 - Qty: 1, Total 50 USD, Discount 20 USD
  - Item 3 - Qty: 1, Total 50 USD, Discount 20 USD
  - Item 4 - Qty: 1, Total 50 USD, Discount 20 USD
  - Item 5 - Qty: 1, Total 50 USD, Discount 20 USD
  
Subtotal = 50 * 5 = 250 USD
Discount total = 20 * 5 = 100 USD
Total = 150 USD

OrderEdit:
- Update Item 1 so that: Qty: 1, Total 50 USD, **Discount remains 20 USD**

Resulting Order:
- Discount (100 USD fixed)
- Items:
  - Item 1 - Qty: 2, Total 100 USD, Discount 20 USD
  - Item 2 - Qty: 1, Total 50 USD, Discount 20 USD
  - Item 3 - Qty: 1, Total 50 USD, Discount 20 USD
  - Item 4 - Qty: 1, Total 50 USD, Discount 20 USD
  - Item 5 - Qty: 1, Total 50 USD, Discount 20 USD

Subtotal = 50 * 4 + 100 * 1 = 300 USD
Discount total = 20 * 5 = 100 USD
Total = 200 USD

^^ The important thing here is that the fixed discount amount is the same before and after the OrderEdit.

Reply:
Another way of putting this:

- If order has percentage discount -> Recompute adjustment for ItemChange
- If order has fixed discount -> Don't do anything; use the original Item adjustment

Reply:
got it 👍 one question about the ids, did you see my comment on the getTotals method?

Reply:
Note: when reducing quantities we might bump into some additional issues e.g.:

### Original Order
- Discount (100 USD fixed)
- Items:
  - Item 1 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 2 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 3 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 4 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 5 - Qty: 2, Total 20 USD, Discount 20 USD
  
Subtotal = 20 * 5 = 100 USD
Discount total = 20 * 5 = 100 USD
Total = 0 USD

### OrderEdit
- Update Item 1 so that: Qty: 1, Total 15 USD, (**Discount remains 20 USD**??)

### Resulting Order A
- Discount (100 USD fixed)
- Items:
  - Item 1 - Qty: 1, Total 10 USD, Discount 20 USD **The total here is now -10 USD; should we allow that?**
  - Item 2 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 3 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 4 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 5 - Qty: 2, Total 20 USD, Discount 20 USD
  
Subtotal = 20 * 4 + 10 * 1  = 90 USD
Discount total = 20 * 5 = 100 USD
Total = -10 USD - this seems odd

^^ usually when you have a cart with subtotal X and a fixed discount with total Y > X then the total is just always 0

So maybe the right approach here would be to have this:

### Resulting Order B
- Discount (100 USD fixed)
- Items:
  - Item 1 - Qty: 1, Total 10 USD, Discount 10 USD
  - Item 2 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 3 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 4 - Qty: 2, Total 20 USD, Discount 20 USD
  - Item 5 - Qty: 2, Total 20 USD, Discount 20 USD

Reply:
yes I was trying to figured out an approach here, in case the quantity is reduced compared to the original item, I think we should re compute the adjustments. If the quantity is greater than the original one then use already existing adjustments in case there is no percentage discounts, otherwise re compute the adjustments which gives the following flow:

- Is line item quantity < original_line_item_quantity
  - Compute adjustments
  - Otherwise
    - Filter on percentage discount, has discounts 
      - YES: Clone original line adjustment to the line item
      - NO: Compute Adjustments

Would that be alright?